pfb/test/test_wavelets.py

```diff
@@ -0,0 +1,275 @@
+    assert py_wave.short_family_name == pfb_wave.short_name
+    assert py_wave.vanishing_moments_phi == pfb_wave.vanishing_moments_phi
+    assert py_wave.vanishing_moments_psi == pfb_wave.vanishing_moments_psi
+
+    assert_array_almost_equal(py_wave.rec_lo, pfb_wave.rec_lo)
+    assert_array_almost_equal(py_wave.dec_lo, pfb_wave.dec_lo)
+    assert_array_almost_equal(py_wave.rec_hi, pfb_wave.rec_hi)
+    assert_array_almost_equal(py_wave.dec_hi, pfb_wave.dec_hi)
+
+
+@pytest.mark.parametrize("wavelet", ["db1", "db4", "db5"])
+@pytest.mark.parametrize("data_shape", [(13,), (12, 7)])
+def test_dwt_idwt_axis(wavelet, data_shape):
+    pywt = pytest.importorskip("pywt")
+    data = np.random.random(size=data_shape)
+    pywt_dwt_axis = pywt._dwt.dwt_axis
+    pywt_idwt_axis = pywt._dwt.idwt_axis
+
+    pywt_wavelet = pywt.Wavelet(wavelet)
+    pywt_mode = pywt.Modes.symmetric
+
+    wavelet = discrete_wavelet(wavelet)
+
+    for axis in reversed(range(len(data_shape))):
+        # Deconstruct
+        ca, cd = dwt_axis(data, wavelet, Modes.symmetric, axis)
+        pywt_ca, pywt_cd = pywt_dwt_axis(data, pywt_wavelet, pywt_mode, axis)
+        assert_array_almost_equal(ca, pywt_ca)
+        assert_array_almost_equal(cd, pywt_cd)
+
+        # Reconstruct with both approximation and detail
+        pywt_out = pywt_idwt_axis(ca, cd, pywt_wavelet, pywt_mode, axis)
+        output = idwt_axis(ca, cd, wavelet, Modes.symmetric, axis)
+        assert_array_almost_equal(output, pywt_out)
+
+        # Reconstruct with approximation only
+        pywt_out = pywt_idwt_axis(ca, None, pywt_wavelet, pywt_mode, axis)
+        output = idwt_axis(ca, None, wavelet, Modes.symmetric, axis)
+        assert_array_almost_equal(output, pywt_out)
+
+        # Reconstruct with detail only
+        pywt_out = pywt_idwt_axis(None, cd, pywt_wavelet, pywt_mode, axis)
+        output = idwt_axis(None, cd, wavelet, Modes.symmetric, axis)
+        assert_array_almost_equal(output, pywt_out)
+
+
+def test_dwt_idwt():
+    pywt = pytest.importorskip("pywt")
+    data = np.random.random((5, 8, 11))
+
+    res = dwt(data, "db1", "symmetric")
+    pywt_res = pywt.dwtn(data, "db1", "symmetric")
+    for k, v in res.items():
+        assert_array_almost_equal(v, pywt_res[k])
+
+    res = dwt(data, "db1", "symmetric", 1)
+    pywt_res = pywt.dwtn(data, "db1", "symmetric", (1,))
+    for k, v in res.items():
+        assert_array_almost_equal(v, pywt_res[k])
+
+    res = dwt(data, ("db1", "db2"), ("symmetric", "symmetric"), (0, 1))
+    pywt_res = pywt.dwtn(data, ("db1", "db2"), ("symmetric", "symmetric"), (0, 1))
+    for k, v in res.items():
+        assert_array_almost_equal(v, pywt_res[k])
+
+    output = idwt(res, ("db1", "db2"), ("symmetric", "symmetric"), (0, 1))
+    pywt_out = pywt.idwtn(pywt_res, ("db1", "db2"), ("symmetric", "symmetric"), (0, 1))
+    assert_array_almost_equal(output, pywt_out)
+
+
+@pytest.mark.parametrize("data_shape", [(50, 24, 63)])
```

Comment:
Can we test this with multiple levels as well please?

Reply:
Yes, it should be possible to parametrize this on level.